### DISCUSSION TITLE
How to connect to sharepoint site using app registered in sharepoint usingPnP core

### DISCUSSION THREAD

**Discussion by manjunathreddy1** [pnp/pnpcore#1148] [General]
Hi All,

Can anyone help me to create the code equivalent to the below code in PnP code for connecting to SharePoint using app registered in sharepoint and not in Azure AD. In the samples folder there is only code to connect to SharePoint using app registered in Azure AD.

 PnP.Framework.AuthenticationManager authManager = new PnP.Framework.AuthenticationManager();
            var sharePointAccessToken = authManager.GetACSAppOnlyContext(siteUrl, clientId, clientSecret).GetAccessToken();

**Comment by jansenbe** [comment]
@manjunathreddy1 : we don't have native support Azure ACS in PNP Core SDK, but we support you bringing your own auth provider by implementing the `IAuthenticationProvider` interface. https://pnp.github.io/pnpcore/demos/README.html lists our samples and shows when there's a custom provider, these can be a basis for your custom code.

Furthermore you can copy code from PnP Framework for the ACS bits.

**Reply by jansenbe** [reply]
@manjunathreddy1 : there's a related issue, linking for reference: https://github.com/pnp/pnpcore/issues/1150
